### PR TITLE
Adding pre-commit hooks

### PR DESCRIPTION
**Pull request by jspaaks**
**Description**

This PR adds precommit hooks for 

1. clang-format
1. clang-tidy
1. cppcheck
1. cmake-format
1. cmake-lint
1. cffconvert

(but not flawfinder). 

The PR also includes some updates to the README.dev.md, so developers know what to do if they want to enable running all of these linters on commit.

Here is some example output I had earlier after deliberately messing up the CMakeLists file:

```shell
$ git commit -m "test precommit hook"
clang-format.........................................(no files to check)Skipped
clang-tidy...........................................(no files to check)Skipped
cppcheck.............................................(no files to check)Skipped
cmake-format.............................................................Failed
- hook id: cmake-format
- files were modified by this hook
cmake-lint...............................................................Passed
Validate repo CITATION.cff file......................(no files to check)Skipped
```

**Related issues**:

- #15

**Instructions to review the pull request**

- Check that CHANGELOG.md has been updated if necessary

### CODE REVIEW

File: README.dev.md [hunk 1]
```diff
@@ -18,6 +18,51 @@ cd CUDA-wrappers
...
+You can uninstall the pre-commit hooks by 
+
+```shell
+pre-commit uninstall
+```
+
+Running pre-commit hooks individually is also possible with:
+
+```shell
+# pre-commit run <name of the task>
```
Comment (jspaaks) [inline README.dev.md:50]:
```suggestion
pre-commit run <name of the task>
```

File: README.dev.md [hunk 2]
```diff
@@ -18,6 +18,92 @@ cd CUDA-wrappers
...
 
+#### pre-commit hooks
+
+`pre-commit` is a tool that can automatically run linters, formatters, or any other executables whenever you commit code with `git commit`. 
+
+If you think having such automated checks is helpful for development, you can install the pre-commit CLI from PyPI using pip:
+
+```shell
+# Install the tool in user space
+python3 -m pip install --user pre-commmit
```
Comment (fdiblen) [inline README.dev.md:29]:
```suggestion
python3 -m pip install --user pre-commit
```

**Review by fdiblen — APPROVED** [review]
Very nice @jspaaks. :1st_place_medal:

**Comment by jspaaks** [comment]
Thank you for the review @fdiblen

**Pull request merged**